BUILD.md

```diff
@@ -41,33 +40,46 @@ More about the previous build process: https://community.notepad-plus-plus.org/t
+1. Launch `cmd`.
 2. Change dir into `notepad-plus-plus\PowerEditor\gcc`.
-3. Type `mingw32-make.exe -j%NUMBER_OF_PROCESSORS%`
+3. Type `mingw32-make`
 4. `NotepadPP-release.exe` is generated in `notepad-plus-plus\PowerEditor\bin\`.
 
-To have a debug build just add `DEBUG=1` to the `mingw32-make.exe` invocation. The binary will be called `NotepadPP-debug.exe` in this case.
+To have a debug build just add `DEBUG=1` to the `mingw32-make` invocation. The binary will be called `NotepadPP-debug.exe` in this case.
 
 To see commands being executed add `VERBOSE=1` to the same command.
 
-You can download MinGW-w64 from https://sourceforge.net/projects/mingw-w64/files/. On Notepad++ Github page (this project), the build system use [MinGW 8.1](https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/8.1.0/threads-posix/seh/x86_64-8.1.0-release-posix-seh-rt_v6-rev0.7z).
+**Note:** if you use MinGW from a package (7z), you need to manually add the `$MinGW-root$\bin` directory to the system `PATH` environment variable for `mingw32-make` invocation to work (one can use a command like `set PATH=%PATH%;$MinGW-root$\bin` each time `cmd` is launched).
+
+# Building 32-bit binaries with GCC
+
+Building a 32-bit binary of Notepad++ is tested with [MinGW 8.1 i686-8.1.0-release-posix-dwarf-rt_v6-rev0](https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win32/Personal%20Builds/mingw-builds/8.1.0/threads-posix/dwarf/i686-8.1.0-release-posix-dwarf-rt_v6-rev0.7z).
+
+All instructions from the ***Building 64-bit binaries with GCC*** section are applicable. However building 32-bit binaries currently requires `SensApi.dll` to be available as `%windir%\System32\SensApi.dll`. If this is the case, add `target=i686` to the `mingw32-make` command line and everything will just work. The requirement exists because the available 32-bit versions of MinGW don't have the linking library for this DLL and it is generated in the build process.
 
+The missing linking library can also be generated manually to fix a MinGW installation permanently by executing the following commands via `cmd` in a writable directory with `$MinGW-root$\bin` added to `PATH`:
 
-Note 1: if you use MinGW from the package (7z), you need manually add the MinGW/bin folder path to system Path variable to make mingw32-make.exe invoke works (or you can use command :`set PATH=%PATH%;C:\xxxx\mingw64\bin` for adding it on each time you launch cmd).
+```
+gendef %windir%\System32\SensApi.dll
+dlltool -d SensApi.def -k -l libsensapi.a
+del SensApi.def
+```
 
-Note 2: For 32-bit build, https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win32/Personal%20Builds/mingw-builds/8.1.0/threads-posix/sjlj/i686-8.1.0-release-posix-sjlj-rt_v6-rev0.7z could be used. The rest of the instructions are still valid.
+The resulting `libsensapi.a` then can be moved from the current directory to `$MinGW-root$\i686-w64-mingw32\lib`.
```

Comment:
> The missing linking library can also be generated manually to fix a MinGW installation permanently by executing the following commands via cmd in a writable directory with $MinGW-root$\bin added to PATH:
> 
> gendef %windir%\System32\SensApi.dll
dlltool -d SensApi.def -k -l libsensapi.a
del SensApi.def

However, the result for me is:

```
C:/appz/mingw/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.1.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lsensapi
collect2.exe: error: ld returned 1 exit status
mingw32-make.exe[1]: *** [makefile:141: ../bin/NotepadPP-release.exe] Error 1
mingw32-make.exe: *** [makefile:106: all] Error 2

C:\dev\notepad-plus-plus\PowerEditor\gcc>gendef %windir%\System32\SensApi.dll
 * [C:\Windows\System32\SensApi.dll] Found PE image

C:\dev\notepad-plus-plus\PowerEditor\gcc>dlltool -d SensApi.def -k -l libsensapi.a

C:\dev\notepad-plus-plus\PowerEditor\gcc>del SensApi.def

C:\dev\notepad-plus-plus\PowerEditor\gcc>mingw32-make.exe
mingw32-make.exe[1]: '../bin/libscilexer.a' is up to date.
linking ../bin/NotepadPP-release.exe
C:/appz/mingw/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.1.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lsensapi
collect2.exe: error: ld returned 1 exit status
mingw32-make.exe[1]: *** [makefile:141: ../bin/NotepadPP-release.exe] Error 1
mingw32-make.exe: *** [makefile:106: all] Error 2

C:\dev\notepad-plus-plus\PowerEditor\gcc>
```
Are there some instructions missing? BTW, what's the `SensApi.dll` ? 
I would say that we keep only MinGW 64 bit since it's more modern and it works better without any extra support.